### PR TITLE
[Misc] refresh documentation

### PR DESCRIPTION
**Pull request by sskserk**
Updated documentation.
Mentioned 2 additional engine parameters.

### CODE REVIEW

File: docs/source/models/engine_args.rst [hunk 1]
```diff
@@ -9,6 +9,15 @@ Below, you can find an explanation of every engine argument for vLLM:
 
     Name or path of the huggingface model to use.
 
+.. option:: --served-model-name <server_model_name>
+
+    The model name used in the API. If not specified, the model name will be the same as the huggingface name.
+
+.. api-key:: --api-key <api_key>
```
Comment (ywang96) [inline docs/source/models/engine_args.rst:16]:
```suggestion
.. option:: --api-key <api_key>
```

File: docs/source/models/engine_args.rst [hunk 2]
```diff
@@ -9,6 +9,15 @@ Below, you can find an explanation of every engine argument for vLLM:
 
     Name or path of the huggingface model to use.
 
+.. option:: --served-model-name <server_model_name>
```
Comment (ywang96) [inline docs/source/models/engine_args.rst:12]:
```suggestion
.. option:: --served-model-name <served_model_name>
```

**Review by ywang96 — COMMENTED** [review]
Thank you for contributing to our documentation! We actually have a dedicated section for the Open AI API server CLI args section [here](https://github.com/vllm-project/vllm/blob/main/docs/source/serving/openai_compatible_server.md#command-line-arguments-for-the-server) but looks like it's [currently not working](https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html#command-line-arguments-for-the-server) for some reason. 

I think this will be fixed once we have a proper fix to our documentation CI, and any contribution is very welcomed!

File: docs/source/models/engine_args.rst [hunk 3]
```diff
@@ -9,6 +9,15 @@ Below, you can find an explanation of every engine argument for vLLM:
...
+
+    The model name used in the API. If not specified, the model name will be the same as the huggingface name.
+
+.. api-key:: --api-key <api_key>
+
+    If provided, the server will require this key to be presented in the header.
+    Example: 'Authorization: Bearer api_key'
+
```
[the remark this answers is not in the document]
Comment (sskserk) [reply docs/source/models/engine_args.rst:20]: @ywang96 ,
oh, I see. Thank you for the clarification! 

..let me close my MR as improper.

**Pull request closed without merging**